missing annotations library for testing

after installing `ogr` via ``pip3 install .``
I wanted to run make check ``PYTHONPATH=`pwd` pytest-3 -v ./tests/``

it leads to errors like:
```
E     File "/home/jscotka/git/userspace-containerization/ogr/ogr/abstract.py", line 1
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
```

seems that annotations module is not accessible in ``python3-3.6.8-1.fc28.x86_64`` by default

--- a/ogr/abstract.py
+++ b/ogr/abstract.py
@@ -1,5 +1,3 @@
-from __future__ import annotations
-
 import datetime
 from dataclasses import dataclass
 from enum import Enum
@@ -39,7 +37,7 @@ class GitService:
         pass
 
     @classmethod
-    def create_from_remote_url(cls, remote_url) -> GitService:
+    def create_from_remote_url(cls, remote_url) -> "GitService":
         """
         Create instance of service from provided remote_url.
 
@@ -48,7 +46,7 @@ class GitService:
         """
         raise NotImplementedError()
 
-    def get_project(self, **kwargs) -> GitProject:
+    def get_project(self, **kwargs) -> "GitProject":
         """
         Get the GitProject instance
 
@@ -60,7 +58,7 @@ class GitService:
         raise NotImplementedError
 
     @property
-    def user(self) -> GitUser:
+    def user(self) -> "GitUser":
         """
         GitUser instance for used token.
 
@@ -130,7 +128,7 @@ class GitProject:
         """
         raise NotImplementedError()
 
-    def get_fork(self) -> Optional[GitProject]:
+    def get_fork(self) -> Optional["GitProject"]:
         """
         GitProject instance of the fork if the fork exists, else None
 
@@ -138,7 +136,7 @@ class GitProject:
         """
         raise NotImplementedError()
 
-    def get_pr_list(self, status: PRStatus = PRStatus.open) -> List[PullRequest]:
+    def get_pr_list(self, status: PRStatus = PRStatus.open) -> List["PullRequest"]:
         """
         List of pull requests (dics)
 
@@ -147,7 +145,7 @@ class GitProject:
         """
         raise NotImplementedError()
 
-    def get_pr_info(self, pr_id: int) -> PullRequest:
+    def get_pr_info(self, pr_id: int) -> "PullRequest":
         """
         Get pull request info
 
@@ -156,7 +154,7 @@ class GitProject:
         """
         raise NotImplementedError()
 
-    def _get_all_pr_comments(self, pr_id: int) -> List[PRComment]:
+    def _get_all_pr_comments(self, pr_id: int) -> List["PRComment"]:
         """
         Get list of pull-request comments.
 
@@ -166,8 +164,8 @@ class GitProject:
         raise NotImplementedError()
 
     def get_pr_comments(
-            self, pr_id, filter_regex: str = None, reverse: bool = False
-    ) -> List[PRComment]:
+        self, pr_id, filter_regex: str = None, reverse: bool = False
+    ) -> List["PRComment"]:
         """
         Get list of pull-request comments.
 
@@ -179,11 +177,11 @@ class GitProject:
         raise NotImplementedError()
 
     def search_in_pr(
-            self,
-            pr_id: int,
-            filter_regex: str,
-            reverse: bool = False,
-            description: bool = True,
+        self,
+        pr_id: int,
+        filter_regex: str,
+        reverse: bool = False,
+        description: bool = True,
     ) -> Optional[Match[str]]:
         """
         Find match in pull-request description or comments.
@@ -197,8 +195,8 @@ class GitProject:
         raise NotImplementedError()
 
     def pr_create(
-            self, title: str, body: str, target_branch: str, source_branch: str
-    ) -> PullRequest:
+        self, title: str, body: str, target_branch: str, source_branch: str
+    ) -> "PullRequest":
         """
         Create a new pull request.
 
@@ -211,13 +209,13 @@ class GitProject:
         raise NotImplementedError()
 
     def pr_comment(
-            self,
-            pr_id: int,
-            body: str,
-            commit: str = None,
-            filename: str = None,
-            row: int = None,
-    ) -> PRComment:
+        self,
+        pr_id: int,
+        body: str,
+        commit: str = None,
+        filename: str = None,
+        row: int = None,
+    ) -> "PRComment":
         """
         Add new comment to the pull request.
 
@@ -230,7 +228,7 @@ class GitProject:
         """
         raise NotImplementedError()
 
-    def pr_close(self, pr_id: int) -> PullRequest:
+    def pr_close(self, pr_id: int) -> "PullRequest":
         """
         Close the pull-request.
 
@@ -239,7 +237,7 @@ class GitProject:
         """
         raise NotImplementedError()
 
-    def pr_merge(self, pr_id: int) -> PullRequest:
+    def pr_merge(self, pr_id: int) -> "PullRequest":
         """
         Merge the pull request.
 
@@ -279,7 +277,7 @@ class GitProject:
 
 
 class GitUser:
-    def __init__(self, service: GitService) -> None:
+    def __init__(self, service: "GitService") -> None:
         self.service = service
 
     def get_username(self) -> str:

--- a/ogr/abstract.py
+++ b/ogr/abstract.py
@@ -1,7 +1,6 @@
 import datetime
-from dataclasses import dataclass
 from enum import Enum
-from typing import Optional, Match, List, Dict, AnyStr
+from typing import Optional, Match, List, Dict
 
 
 class PRStatus(Enum):
@@ -11,25 +10,42 @@ class PRStatus(Enum):
     all = 4
 
 
-@dataclass
 class PullRequest:
-    title: str
-    id: int
-    status: PRStatus
-    url: str
-    description: str
-    author: str
-    source_branch: str
-    target_branch: str
-    created: datetime.datetime
+    def __init__(
+        self,
+        title: str,
+        id: int,
+        status: PRStatus,
+        url: str,
+        description: str,
+        author: str,
+        source_branch: str,
+        target_branch: str,
+        created: datetime.datetime,
+    ) -> None:
+        self.title = title
+        self.id = id
+        self.status = status
+        self.url = url
+        self.description = description
+        self.author = author
+        self.source_branch = source_branch
+        self.target_branch = target_branch
+        self.created = created
 
 
-@dataclass
 class PRComment:
-    comment: str
-    author: str
-    created: datetime.datetime
-    edited: datetime.datetime
+    def __init__(
+        self,
+        comment: str,
+        author: str,
+        created: datetime.datetime,
+        edited: datetime.datetime,
+    ) -> None:
+        self.comment = comment
+        self.author = author
+        self.created = created
+        self.edited = edited
 
 
 class GitService:

--- a/ogr/services/github.py
+++ b/ogr/services/github.py
@@ -1,5 +1,3 @@
-from __future__ import annotations
-
 import logging
 from typing import Optional, Dict, List
 
@@ -20,8 +18,8 @@ class GithubService(BaseGitService):
         self.github = github.Github(login_or_token=self._token)
 
     def get_project(
-            self, repo=None, namespace=None, is_fork=False, **kwargs
-    ) -> GithubProject:
+        self, repo=None, namespace=None, is_fork=False, **kwargs
+    ) -> "GithubProject":
         if is_fork:
             namespace = self.user.get_username()
         return GithubProject(repo=repo, namespace=namespace, service=self, **kwargs)
@@ -79,7 +77,7 @@ class GithubProject(BaseGitProject):
         ]
 
     def pr_create(
-            self, title: str, body: str, target_branch: str, source_branch: str
+        self, title: str, body: str, target_branch: str, source_branch: str
     ) -> PullRequest:
         created_pr = self.github_repo.create_pull(
             title=title, body=body, base=target_branch, head=source_branch
@@ -87,12 +85,12 @@ class GithubProject(BaseGitProject):
         return self._pr_from_github_object(created_pr)
 
     def pr_comment(
-            self,
-            pr_id: int,
-            body: str,
-            commit: str = None,
-            filename: str = None,
-            row: int = None,
+        self,
+        pr_id: int,
+        body: str,
+        commit: str = None,
+        filename: str = None,
+        row: int = None,
     ) -> PRComment:
         raise NotImplementedError
 

--- a/ogr/services/pagure.py
+++ b/ogr/services/pagure.py
@@ -1,5 +1,3 @@
-from __future__ import annotations
-
 import datetime
 import logging
 from typing import List, Optional, Dict
@@ -14,10 +12,10 @@ logger = logging.getLogger(__name__)
 
 class PagureService(BaseGitService):
     def __init__(
-            self,
-            token: str = None,
-            instance_url: str = "https://src.fedoraproject.org",
-            **kwargs,
+        self,
+        token: str = None,
+        instance_url: str = "https://src.fedoraproject.org",
+        **kwargs,
     ) -> None:
         super().__init__()
         self.instance_url = instance_url
@@ -26,7 +24,7 @@ class PagureService(BaseGitService):
 
         self.pagure = OurPagure(pagure_token=token, instance_url=instance_url, **kwargs)
 
-    def get_project(self, **kwargs) -> PagureProject:
+    def get_project(self, **kwargs) -> "PagureProject":
         project_kwargs = self.pagure_kwargs.copy()
         project_kwargs.update(kwargs)
         return PagureProject(
@@ -37,7 +35,7 @@ class PagureService(BaseGitService):
         )
 
     @property
-    def user(self) -> PagureUser:
+    def user(self) -> "PagureUser":
         return PagureUser(service=self)
 
     def change_token(self, new_token: str) -> None:
@@ -52,15 +50,15 @@ class PagureService(BaseGitService):
 
 class PagureProject(BaseGitProject):
     def __init__(
-            self,
-            repo: str,
-            namespace: str,
-            service: PagureService,
-            username: Optional[str] = None,
-            instance_url: Optional[str] = None,
-            token: Optional[str] = None,
-            is_fork: bool = False,
-            **kwargs,
+        self,
+        repo: str,
+        namespace: str,
+        service: "PagureService",
+        username: Optional[str] = None,
+        instance_url: Optional[str] = None,
+        token: Optional[str] = None,
+        is_fork: bool = False,
+        **kwargs,
     ) -> None:
         if is_fork and username:
             complete_namespace = f"fork/{username}/{namespace}"
@@ -118,12 +116,12 @@ class PagureProject(BaseGitProject):
         return parsed_comments
 
     def pr_comment(
-            self,
-            pr_id: int,
-            body: str,
-            commit: str = None,
-            filename: str = None,
-            row: int = None,
+        self,
+        pr_id: int,
+        body: str,
+        commit: str = None,
+        filename: str = None,
+        row: int = None,
     ) -> PRComment:
         return self._pagure.comment_request(
             request_id=pr_id, body=body, commit=commit, filename=filename, row=row
@@ -136,7 +134,7 @@ class PagureProject(BaseGitProject):
         return self._pagure.merge_request(request_id=pr_id)
 
     def pr_create(
-            self, title: str, body: str, target_branch: str, source_branch: str
+        self, title: str, body: str, target_branch: str, source_branch: str
     ) -> PullRequest:
         pr_info = self._pagure.create_request(
             title=title,
@@ -150,7 +148,7 @@ class PagureProject(BaseGitProject):
     def fork_create(self) -> None:
         self._pagure.create_fork()
 
-    def get_fork(self) -> Optional[PagureProject]:
+    def get_fork(self) -> Optional["PagureProject"]:
         """
         PagureRepo instance of the fork of this repo.
         """
@@ -237,7 +235,7 @@ class PagureProject(BaseGitProject):
 
 
 class PagureUser(BaseGitUser):
-    def __init__(self, service: PagureService) -> None:
+    def __init__(self, service: "PagureService") -> None:
         super().__init__(service=service)
 
     def get_username(self) -> str:

